ClientApp/viking-vault/src/UserDashboard/AccountPage.tsx

```diff
@@ -170,6 +186,50 @@ class AccountPage extends React.Component<any, IAccountState>{
         return this.state.cardNumber.substring(begin, end);
     }
 
+    blockCard = () =>{
+        const token = sessionStorage.getItem('Authentication-Token');
+        if(token === null) return;
+        fetch(constants.baseUrl+"card", {
+            method: "PUT",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token': token
+            },
+            body: JSON.stringify({
+              Id: this.state.cardId,
+              CardNumber: this.state.cardNumber,
+              ExpirationDate: this.state.expirationDate,
+              CCV: this.state.CCV,
+              Blocked: !this.state.isCardBlocked
+            })}).then(response => {   
+                if(response.status === 200){
+                    let oldState = this.state.isCardBlocked;
+                    this.setState({
+                        isCardBlocked : !oldState
+                    });
+                }
+                if(response.status !== 200) {
```

Comment:
you can all this.openResponseModalWithMessage("Something wrong happened. Try again later!") on **else** branch of    if(response.status === 200) instead of verify again if it's different than 200